contributions/catalog/models/capability_utilities.py

```diff
@@ -63,14 +64,15 @@ def to_capability(d):
         # init capability
         for _ in range(num_cap):
             capability_list.append(init_capability())
-
     for i, capability in enumerate(capability_list):
         cap_id = str(uuid.uuid4())
         capability['id'] = cap_id
-
-        env_k, env_v = d['environmentVariables_key_' + str(i)], d['environmentVariables_value_' + str(i)]
-        for k, v in list(zip(env_k, env_v)):
-            capability["deploymentDetails"]['environmentVariables'].append({'key': k, 'value': v})
+        key_pattern = re.compile('environmentVariables_key_[0-9]+' + '_' + str(i))
+        val_pattern = re.compile('environmentVariables_value_[0-9]+' + '_' + str(i))
+        d_keys = list(filter(key_pattern.match, d))  # filter keys matching pattern
+        d_vals = list(filter(val_pattern.match, d))  # filter keys matching pattern
+        for k, v in zip(d_keys, d_vals):
```

Comment:
Please add some comments explaining the logic in this method. Thanks.

Reply:
updated